lua/autorun/server/sv_limit_explosives.lua

```diff
@@ -28,25 +36,26 @@ local function onExplosiveCreated( owner, ent )
 
 end
 
-local function onExplosiveRemoved( owner, ent )
+local function onExplosiveRemoved( ent )
+    if not IsValid( ent ) then return end
+    if not limits[ent:GetClass()] then return end
+    local owner = ent:GetInternalVariable( "m_hThrower" ) or ent.Owner
+
     if not IsValid( owner ) then return end
     if not IsValid( ent ) then return end
 
     local explosives = getExplosivesTable( owner, ent:GetClass() )
     if explosives == nil then return end
+
     table.RemoveByValue( explosives, ent )
 end
 
 hook.Remove( "OnEntityCreated", "CFC_LimitExplosives" )
 hook.Add( "OnEntityCreated", "CFC_LimitExplosives", function( ent )
-    if limits[ent:GetClass()] == nil then return end
```

Comment:
I think it's still safe to check the class here

Reply:
it wasnt because it's ran on a 0 tick timer. for some reason crossbow bolts error it idk why

Reply:
The 0-tick timer makes sure a lot of stuff on the entity is setup (cppi owner and such), but the class _must_ be defined on created

Reply:
[cfc_placed_explosives_limiter] addons/cfc_placed_explosives_limiter/lua/autorun/server/sv_limit_explosives.lua:16: Tried to use a NULL entity!
  1. GetClass - [C]:-1
   2. onExplosiveCreated - addons/cfc_placed_explosives_limiter/lua/autorun/server/sv_limit_explosives.lua:16
    3. unknown - addons/cfc_placed_explosives_limiter/lua/autorun/server/sv_limit_explosives.lua:55

Timer Failed! [Simple][@addons/cfc_placed_explosives_limiter/lua/autorun/server/sv_limit_explosives.lua (line 54)]